packages/MJCore/src/generic/baseEntity.ts

```
@@ -628,6 +712,29 @@ export abstract class BaseEntity<T = unknown> {
             if (field.EntityFieldInfo.TSType === EntityFieldTSType.Date && (typeof Value === 'string' || typeof Value === 'number') ) {
                 field.Value = new Date(Value);
             }
+            else if (field.EntityFieldInfo.TSType === EntityFieldTSType.Number && typeof Value === 'string' && Value !== null && Value !== undefined) {
+                const numericValue = Number(Value);
+                if (!isNaN(numericValue)) {
+                    field.Value = numericValue;
+                } else {
+                    field.Value = Value;
+                }
+            }
+            else if (field.EntityFieldInfo.TSType === EntityFieldTSType.Boolean && Value !== null && Value !== undefined) {
+                if (typeof Value === 'string') {
+                    if (Value.trim() === '1' || Value.trim().toLowerCase() === 'true') {
+                        field.Value = true;
+                    } else if (Value.trim() === '0' || Value.trim().toLowerCase() === 'false') {
+                        field.Value = false;
+                    } else {
+                        field.Value = Value;
+                    }
+                } else if (typeof Value === 'number') {
+                    field.Value = Value !== 0;
+                } else {
+                    field.Value = Value;
```

in an odd case if someone set a Boolean field to a Date or complex object or array - this would directly set it. I think in those cases perhaps we throw an exception instead of silently converting it?